orchard_sinsemilla.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python3
+import sys; assert sys.version_info[0] >= 3, "Python 3 required."
+
+import math
+
+import orchard_iso_pallas
+
+from pyblake2 import blake2b, blake2s
+from orchard_pallas import Fp, p, q, PALLAS_B
+from orchard_iso_pallas import PALLAS_ISO_B, PALLAS_ISO_A
+from sapling_utils import i2beosp, cldiv, beos2ip
+from binascii import hexlify
+
+# https://stackoverflow.com/questions/2612720/how-to-do-bitwise-exclusive-or-of-two-strings-in-python
+def sxor(s1,s2):    
+    # convert strings to a list of character pair tuples
+    # go through each tuple, converting them to ASCII code (ord)
+    # perform exclusive or on the ASCII code
+    # then convert the result back to ASCII (chr)
+    # merge the resulting array of characters as a string
+    return bytes([a ^ b for a,b in zip(s1,s2)])
+
+def expand_message_xmd(msg, dst, len_in_bytes):
+
+    b_in_bytes = 64
+
+    ell = cldiv(len_in_bytes, b_in_bytes)
+    if ell > 255:
+        raise
+    dst_prime = dst + i2beosp(8, len(dst))  # check
+    r_in_bytes = 64  # check
```

Comment:
```suggestion
    r_in_bytes = 128
```
This was an error in the spec pointed out by NCC; see https://github.com/zcash/pasta/pull/2